zboxcore/sdk/attributesworker.go

```diff
@@ -43,63 +43,39 @@ func (ar *AttributesRequest) getObjectTreeFromBlobber(
 func (ar *AttributesRequest) updateBlobberObjectAttributes(
 	blobber *blockchain.StorageNode, blobberIdx int) (
 	re fileref.RefEntity, err error) {
+	var s strings.Builder
 
 	re, err = ar.getObjectTreeFromBlobber(ar.blobbers[blobberIdx])
 	if err != nil {
 		return
 	}
 
-	var (
-		body bytes.Buffer
-		form = multipart.NewWriter(&body)
-	)
-
-	form.WriteField("connection_id", ar.connectionID)
-	form.WriteField("path", ar.remotefilepath)
-	form.WriteField("attributes", ar.attributes)
-
-	form.Close()
+	remoteFilePathHash := fileref.GetReferenceLookup(ar.allocationID, ar.remotefilepath)
 
-	var httpreq *http.Request
-	httpreq, err = zboxutil.NewAttributesRequest(blobber.Baseurl,
-		ar.allocationTx, &body)
+	respRaw, err := blobberClient.UpdateObjectAttributes(blobber.Baseurl, &blobbergrpc.UpdateObjectAttributesRequest{
+		Path:         ar.remotefilepath,
+		PathHash:     remoteFilePathHash,
+		Allocation:   ar.allocationTx,
+		ConnectionId: ar.connectionID,
+		Attributes:   ar.attributes,
+	})
 	if err != nil {
-		Logger.Error(blobber.Baseurl,
-			"Error creating update attributes request", err)
+		Logger.Error("could not update object attributes from blobber -" + blobber.Baseurl + " - " + err.Error())
 		return
 	}
+	s.WriteString(string(respRaw))
 
-	httpreq.Header.Add("Content-Type", form.FormDataContentType())
-
-	var ctx, cncl = context.WithTimeout(ar.ctx, (time.Second * 30))
-	defer cncl()
-
-	err = zboxutil.HttpDo(ctx, cncl, httpreq,
-		func(resp *http.Response, err error) error {
-			if err != nil {
-				Logger.Error("Request error: ", err)
-				return err
-			}
-			defer resp.Body.Close()
-
-			if resp.StatusCode == http.StatusOK {
-				ar.consensus++
-				ar.attributesMask |= (1 << uint32(blobberIdx))
-				Logger.Info(blobber.Baseurl, " "+ar.remotefilepath,
-					" attributes updated.")
-				return nil
-			}
-
-			var respBody []byte
-			if respBody, err = ioutil.ReadAll(resp.Body); err != nil {
-				Logger.Error(blobber.Baseurl, "Reading response: ", err)
-				return nil
-			}
+	updateBlobberObjectAttributesResult := &blobberhttp.UpdateObjectAttributesResponse{}
+	err = json.Unmarshal(respRaw, updateBlobberObjectAttributesResult)
```

Comment:
Is there a need for this, we are anyway just updating the objectAttribute & showing if it was successfull or not?

Reply:
yes, i think we can remove this altogether.